utils/manifest-utils/src/policy.rs

```diff
@@ -193,22 +256,175 @@ impl Keystore {
+        Ok(store.permissions.get(cert))
+    }
+
+    fn has_permissions(
+        &self,
+        cert: &X509Ref,
+        required: &Vec<CertPermissions>,
+    ) -> anyhow::Result<()> {
+        let cert = self.get_permissions(cert)?;
+
+        if cert.contains(&CertPermissions::All) && (!required.contains(&CertPermissions::UnverifiedPermissionsChain) || (required.contains(&CertPermissions::UnverifiedPermissionsChain)
+                    && cert.contains(&CertPermissions::UnverifiedPermissionsChain))) {
+            return Ok(());
+        }
+
+        if required.iter().all(|permission| cert.contains(permission)) {
+            return Ok(());
+        }
+
+        bail!(
+            "Not sufficient permissions. Required: `{}`, but has only: `{}`",
+            format_permissions(required),
+            format_permissions(&cert)
+        )
+    }
+
+    fn find_issuer(&self, cert: &X509) -> anyhow::Result<X509> {
+        let store = self
+            .inner
+            .read()
+            .map_err(|err| anyhow::anyhow!("RwLock error: {}", err.to_string()))?;
+        store
+            .store
+            .objects()
+            .iter()
+            .filter_map(|cert| cert.x509())
+            .map(|cert| cert.to_owned())
+            .find(|trusted| trusted.issued(cert) == X509VerifyResult::OK)
+            .ok_or_else(|| anyhow!("Issuer certificate not found in keystore"))
+    }
+
+    fn decode_cert<S: AsRef<str>>(cert: S) -> anyhow::Result<X509> {
+        let cert = crate::decode_data(cert)?;
+        Ok(match X509::from_der(&cert) {
+            Ok(cert) => cert,
+            Err(_) => X509::from_pem(&cert)?,
+        })
+    }
+
+    pub fn permissions_manager(&self) -> PermissionsManager {
+        self.inner.read().unwrap().permissions.clone()
+    }
+}
+
+impl PermissionsManager {
+    pub fn load(cert_dir: impl AsRef<Path>) -> anyhow::Result<PermissionsManager> {
+        let path = cert_dir.as_ref().join(PERMISSIONS_FILE);
+        let content = match std::fs::read_to_string(path) {
+            Ok(content) if !content.is_empty() => content,
+            _ => return Ok(Default::default()),
+        };
+        let permissions = serde_json::from_str(&content)?;
+        Ok(PermissionsManager { permissions })
+    }
+
+    pub fn set(&mut self, cert: &str, mut permissions: Vec<CertPermissions>) {
+        if permissions.contains(&CertPermissions::All) {
+            permissions.clear();
+            permissions.push(CertPermissions::All);
+        }
+        self.permissions.insert(cert.to_string(), permissions);
+    }
+
+    pub fn set_x509(
+        &mut self,
+        cert: &X509,
+        permissions: Vec<CertPermissions>,
+    ) -> anyhow::Result<()> {
+        let id = cert_to_id(cert)?;
+        self.set(&id, permissions);
+        Ok(())
+    }
+
+    pub fn set_many(
+        &mut self,
+        // With slice I would need add `openssl` dependency directly to ya-rovider.
+        #[allow(clippy::ptr_arg)]
```

Comment:
As in comment - a bit late for refactors, and it is better to suppress clippy than to add explicit dependency only to `collect::<Vec<openssl::x509::X509>()`